contracts/SportMarkets/Rundown/GamesOddsObtainer.sol

```diff
@@ -0,0 +1,609 @@
+                consumer.pauseOrUnpauseMarket(mainMarketTotalChildMarket[_main][_game.totalOver], false);
+                consumer.pauseOrUnpauseMarket(currentActiveTotalChildMarket[_main], true);
+                _setCurrentChildMarkets(_main, mainMarketTotalChildMarket[_main][_game.totalOver], false);
+            }
+        }
+    }
+
+    function _createMarketSpreadTotalMarket(
+        bytes32 _gameId,
+        address _mainMarket,
+        bool _isSpread,
+        int16 _spreadHome,
+        uint24 _totalOver
+    ) internal {
+        // create
+        sportsManager.createMarket(
+            _gameId,
+            _append(_gameId, _isSpread, _spreadHome, _totalOver), // gameLabel
+            consumer.getGameCreatedById(_gameId).startTime, //maturity
+            0, //initialMint
+            2, // always two positions for spread/total
+            _calculateTags(consumer.sportsIdPerGame(_gameId), _isSpread), //tags
+            true // is child
+        );
+
+        address _childMarket = sportsManager.getActiveMarketAddress(sportsManager.numActiveMarkets() - 1);
+
+        // adding child markets
+        _setChildMarkets(_gameId, _mainMarket, _childMarket, _isSpread, _spreadHome, _totalOver);
+    }
+
+    function _calculateTags(uint _sportsId, bool _isSpread) internal view returns (uint[] memory) {
+        uint[] memory result = new uint[](2);
+        result[0] = MIN_TAG_NUMBER + _sportsId;
+        result[1] = _isSpread ? TAG_NUMBER_SPREAD : TAG_NUMBER_TOTAL;
+        return result;
+    }
+
+    // "+4.5 - homeTeam vs awayTeam" UP DOWN
+    // "+200 - homeTeam vs awayTeam" UP DOWN
+    function _append(
+        bytes32 _gameId,
+        bool _isSpread,
+        int16 _spreadHome,
+        uint24 _totalOver
+    ) internal view returns (string memory) {
+        return
+            _isSpread
+                ? string(
+                    abi.encodePacked(
+                        _parseSpread(_spreadHome),
+                        " - ",
+                        consumer.getGameCreatedById(_gameId).homeTeam,
+                        " vs ",
+                        consumer.getGameCreatedById(_gameId).awayTeam
+                    )
+                )
+                : string(
+                    abi.encodePacked(
+                        Strings.toString(_totalOver),
+                        " - ",
+                        consumer.getGameCreatedById(_gameId).homeTeam,
+                        " vs ",
+                        consumer.getGameCreatedById(_gameId).awayTeam
+                    )
+                );
+    }
+
+    function _parseSpread(int16 _spreadHome) internal view returns (string memory) {
+        return
+            _spreadHome > 0
+                ? Strings.toString(uint16(_spreadHome))
+                : string(abi.encodePacked("-", Strings.toString(uint16(_spreadHome * (-1)))));
+    }
+
+    function _pauseAllMarkets(
```

Comment:
declares to return bool, but doesnt return anything

Reply:
typo, removed